lib/database_dumper.rb

```diff
@@ -0,0 +1,148 @@
+require 'csv'
+require 'rubygems'
+require 'fileutils'
+
+class DatabaseDumper
+  EXPORT_DIR = Rails.root.join("db", "backup")
+
+  def self.ensure_dir
+    FileUtils.mkdir_p(EXPORT_DIR)
+  end
+
+  def self.cleanup
+    FileUtils.rm_rf(EXPORT_DIR)
+  end
+
+  def self.csv_path(name)
+    File.join(EXPORT_DIR, "#{name}.csv")
+  end
+
+  def self.export_all_csvs
+    ensure_dir
+    save_profiles_csv
+
+    [AuthenticationToken, Authentication, Doorkeeper::Application, Authorization, Feedback,
+     Notification, Doorkeeper::AccessGrant, Doorkeeper::AccessToken, Role, SmsCode, Task,
+     TaskItem, UnsubscribeToken, UserAction].each do |klass|
+      save_csv(klass)
+    end
+
+    save_csv(User, methods: [:role_ids])
+  end
+
+  def self.save_csv(klass, options = {})
+    return unless klass.count > 0  
+
+    column_names = klass.columns.map(&:name)
+    csv_columns = column_names
+    csv_columns += options[:methods] if options[:methods]
+
+    table_name = klass.table_name
+
+    CSV.open(csv_path(table_name), "w") do |csv|
+      csv << csv_columns
+
+      klass.find_each do |x|
+        # this is writing out the typecast Rails versions of the database columns
+        row = column_names.map { |c| x.read_attribute(c) }
+        row += options[:methods].map { |c| x.send(c).join(",") } if options[:methods]
+        csv << row
+      end
+    end
+  end
+
+  def self.save_profiles_csv
```

Comment:
Assignment Branch Condition size for save_profiles_csv is too high. [27.29/15]<br>Method has too many lines. [29/20]